src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -201,14 +202,42 @@ internal static IEnumerable<string> GetCommaSeparatedValues(string text)
             return validName;
         }
 
-        private static IEnumerable<TagHelperAttributeDescriptor> GetAttributeDescriptors(Type type)
+        private static IEnumerable<TagHelperAttributeDescriptor> GetAttributeDescriptors(
+            Type type, 
+            ErrorSink errorSink)
         {
-            var properties = type.GetRuntimeProperties().Where(IsValidProperty);
+            var properties = type.GetRuntimeProperties().Where(IsAccessibleProperty);
             var attributeDescriptors = properties.Select(ToAttributeDescriptor);
 
+            attributeDescriptors = attributeDescriptors.Where(
+                descriptor => ValidTagHelperAttributeDescriptor(descriptor, type, errorSink));
+
             return attributeDescriptors;
         }
 
+        private static bool ValidTagHelperAttributeDescriptor(
```

Comment:
Nah :-1: passing around the generated name is ugly and harder to understand. Throwing away the `TagHelperAttributeDescriptor` only occurs on pages that can't run anyways so it's not a problem.